### PR TITLE
Reginacompton/ch730/handle all required fields

### PR DESCRIPTION
**Pull request by reginafcompton**
# Description

This PR handles required fields in the Pathways/Schems.org definitions, which previously had not been handled.

Closes https://app.clubhouse.io/brighthive/story/730/handle-all-required-fields

# Checklists

### Basic

- [x] Did you write tests for the code in this PR?
- [x] Did you document your changes in the README and/or in docstrings (as needed)?

# Notes for the Reviewer

@loganripplinger - I'd appreciate a review of the changes in the `Converter` class and the helper function `add_identifier_data`. Specifically:

* `_check_for_required` – this function previously checked for the existence of keywords, but it did not check if those keywords had values. I added a check for empty strings, too. Why? Apparently, if a required field has an empty string as the value, then Google pathways raises an error. https://github.com/brighthive/google-pathways-converter/pull/12/files#diff-b9ee80b83caa7233f65166716c9cc6eeR18
* `add_identifier_data` – similar to above, Google pathways raises an error if the identifier field is missing a CIP and/or program ID. I tried to add this check to `_check_for_required`, but it go messy, since the kwargs can include either/or.

### CODE REVIEW

File: converter/converter.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 from converter.helper import add_basic_keywords, add_data_keywords
 
 class Converter():
-    def __init__(self, data_keywords_mapper, kwarg_to_schema_key_mapper, required_keywords):
-        self.required_keywords = ["provider_address", "program_name", "program_url"] + required_keywords
+    def __init__(self, data_keywords_mapper, kwarg_to_schema_key_mapper, additional_required_keywords=[]):
+        self.required_keywords = ["provider_address", "program_name", "program_description", "program_url"] + additional_required_keywords
```
Comment (loganripplinger) [inline converter/converter.py:5]: To prevent an error from occurring if users don't pass additional_required_keywords as a list, could we add a check like
```python
if type(additional_required_keywords) is str:
    additional_required_keywords = [additional_required_keywords]
```

I don't think we're exporting the converter class so maybe it doesn't matter. Up to you.

Comment (loganripplinger) [reply converter/converter.py:5]: I think this only applies when someone extends the library to support another format.

File: converter/converter.py [hunk 2]
```diff
@@ -16,12 +16,21 @@ def trigger_conversion(self, kwargs):
...
+        '''
+        This function checks two things:
+        (1) the kwargs contain all required fields
+        (2) all required fields have a value – Google pathways does not consider an empty string to be valid input.
+        '''
+        missing_kwargs = []
+        for kwarg in self.required_keywords:
+            if kwarg not in kwargs.keys():
+                missing_kwargs.append(kwarg)
+            elif kwargs[kwarg] == '' or kwargs[kwarg] == None:
```
Comment (loganripplinger) [inline converter/converter.py:28]: Do we have/did you add a unit test/regression test to cover this case when a kwarg is empty?

Comment (reginafcompton) [reply converter/converter.py:31]: Ah, I do not. Just empty strings. Will add!

File: converter/converter.py (hunk 1 above)
Comment (reginafcompton) [reply converter/converter.py]: I think this check makes sense, and I like giving the user alternative ways to pass `required_keywords`. I added it over here: https://github.com/brighthive/google-pathways-converter/pull/12/commits/ad66664ea8738cb5f7aed5dc377de3124b2d3584

N.b., I also undid a previous change, from an earlier commit in this PR:
```
self.required_keywords = ["provider_address", "program_name", "program_description", "program_url"] + additional_required_keywords
```
I can imagine having program types beyond WorkBased and EducationalOccupational, which do not require these fields. Sooooo.......extensibility!

**Comment by reginafcompton** [comment]
@loganripplinger - thanks for the review! I made a couple changes per your suggestions. Let me know if you "approve."

**Comment by loganripplinger** [comment]
👍
